osu.Game.Rulesets.Osu.Tests/TestCaseSliderInput.cs

```diff
@@ -0,0 +1,410 @@
+                };
+
+                performStaticInputTest(frames);
+            });
+
+            AddUntilStep(() => allJudgedFired, "Wait for test 4");
+            AddAssert("Tracking retained", assertGreatJudge);
+        }
+
+        /// <summary>
+        ///     Pressing a key before a slider and holding the slider throughout the body should result in tracking, but a miss on the slider head.
+        ///     Only one frame is required:
+        ///     Frame 1: Left Click
+        ///     In a successful test case:
+        ///     The head of the slider should be judged as a miss.
+        ///     The slider end should be judged as a meh.
+        /// </summary>
+        [Test]
+        public void TestTrackingPreclicked()
+        {
+            AddStep("Tracking retention test", () =>
+            {
+                var frames = new List<ReplayFrame>
+                {
+                    new OsuReplayFrame { Position = new Vector2(0, 0), Actions = { OsuAction.LeftButton }, Time = 250},
+                };
+
+                performStaticInputTest(frames);
+            });
+
+            AddUntilStep(() => allJudgedFired, "Wait for test 5");
+            AddAssert("Tracking retained, sliderhead miss", assertHeadMissTailMeh);
+        }
+
+        /// <summary>
+        ///     Hitting a slider head, leaving the slider, then coming back into the slider with the same button to track it should re-start tracking.
+        ///     Only one frame is required:
+        ///     Frame 1: Left Click
+        ///     In a successful test case:
+        ///     The last tick of the slider should be judged as a great.
+        /// </summary>
+        [Test]
+        public void TestTrackingReturnMidSlider()
+        {
+            AddStep("Mid-sldier tracking re-acquisition", () =>
+            {
+                var frames = new List<ReplayFrame>
+                {
+                    new OsuReplayFrame { Position = new Vector2(0, 0), Actions = { OsuAction.LeftButton }, Time = 1500},
+                    new OsuReplayFrame { Position = new Vector2(150, 150), Actions = { OsuAction.LeftButton }, Time = 2000},
+                    new OsuReplayFrame { Position = new Vector2(200, 200), Actions = { OsuAction.LeftButton }, Time = 2500},
+                    new OsuReplayFrame { Position = new Vector2(0, 0), Actions = { OsuAction.LeftButton }, Time = 3000},
+                    new OsuReplayFrame { Position = new Vector2(0, 0), Actions = { OsuAction.LeftButton }, Time = 3500},
+                };
+
+                performStaticInputTest(frames);
+            });
+
+            AddUntilStep(() => allJudgedFired, "Wait for test 6");
+            AddAssert("Tracking re-acquired", assertMidSliderJudgements);
+        }
+
+        /// <summary>
+        ///     Pressing a key before a slider, hitting a slider head, leaving the slider, then coming back into the slider to track it should NOT start retracking
```

Comment:
wrong documentation. need to mention hitting with a second key and also releasing the key?

Reply:
The releasing key test is test 9, but yeah need to mention the hitting second key part